app/controllers/get_involved_controller.rb

```diff
@@ -37,50 +37,70 @@ def retrieve_given_document_type(document_type)
-    cl_docs.sort_by! { |k| -k['details']['closing_date'] }.reverse!
+    query = {
+      filter_display_type: "Closed consultation",
+      filter_end_date: "from: #{cutoff_date}",
+      count: 0,
+    }
 
-    cl_docs.each do |doc|
-      if DateTime.parse(doc['details']['closing_date']) > DateTime.now.prev_month(months)
-        closed_count += 1
-      else
-        break
-      end
-    end
-
-    return closed_count # hard return for clarity
+    Services.search_api.search(query)["total"]
   end
 
   def retrieve_next_closing
-    open_consults = retrieve_given_document_type("open_consultation")['results']
-    open_consults.sort_by! { |k| k['details']['closing_date'] }[0]
+    # Ensure that on query we're not looking in the past
+    cutoff_date = Time.zone.now.to_date
+
+    query = {
+      filter_display_type: "Open consultation",
+      filter_end_date: "from: #{cutoff_date}",
+      fields: "end_date,title,link",
+      order: "end_date",
+      count: 1,
+    }
+
+    Services.search_api.search(query)["results"].first
   end
 
   def retrieve_new_consultations
-    open_consults = retrieve_given_document_type("open_consultation")['results']
-    sorted_desc = open_consults.sort_by! { |k| k['details']['opening_date'] }.reverse!.values_at(0..2)
-    parse_organisation_acronyms(sorted_desc)
+    query = {
+      filter_display_type: "Open consultation",
+      fields: "end_date,title,link,organisations",
+      order: "-start_date",
+      count: 3,
+    }
+
+    Services.search_api.search(query)["results"]
   end
 
   def retrieve_consultation_outcomes
-    closed_consults = retrieve_given_document_type("consultation_outcome")['results']
-    sorted_desc = closed_consults.sort_by! { |k| k['details']['closing_date'] }.reverse!.values_at(0..2)
-    parse_organisation_acronyms(sorted_desc)
+    # Ensure that on query we're not looking into the future
+    cutoff_date = Time.zone.now.to_date
+
+    query = {
+      filter_display_type: "Closed consultation",
+      filter_end_date: "to: #{cutoff_date}",
+      fields: "end_date,title,link,organisations",
+      order: "-end_date",
+      count: 3,
+    }
+
+    Services.search_api.search(query)["results"]
   end
 
-  def parse_organisation_acronyms(consultations)
-    consultations.each do |consultation|
-      org_acronyms = []
-      organisations = consultation['links']['organisations']
-      organisations.each do |org_id|
-        org_acronyms << Services.publishing_api.get_live_content(org_id).parsed_content['details']['acronym']
-      end
-      consultation['links']['organisation_acronyms'] = org_acronyms
+  def retrieve_take_part_pages
+    detailed_results = []
+    results = Services.search_api.search(filter_format: "take_part", order: "ordering", count: 100)["results"]
+
+    results.each do |r| 
+      detailed_results << Services.content_store.content_item(r["link"]).to_h
```

Comment:
I'm a little concerned by the number of API calls here and the nested nature of them meaning (whilst we only have 18 at the moment) the number of calls could become rather large.

Instead I'd suggest using [link expansion](https://github.com/alphagov/publishing-api/blob/master/docs/link-expansion.md) (as recommended by @thomasleese in [this comment](https://github.com/alphagov/government-frontend/pull/1939#discussion_r555845242)).  This would mean you don't need to put the ordering into `search-api` at all.  You'd need to add the content IDs for these into the `links` section of the `/government/get-involved` content item when Whitehall Publisher presents the content item to `publishing-api`.  Publishing API will then expand the links so all the information you need is in the one content item, so no need to make a request to Search API or the content store at all as you're already retrieving the content item for the page.  This basically moves the lookup stuff to the point of publishing, rather than doing it at the point of rendering the page to a user.

You can see an example of how this is used on organisation pages, where contacts are added as `links` to the content item (e.g. [for the AAIB](https://www.gov.uk/api/content/government/organisations/air-accidents-investigation-branch)).